src/coreclr/vm/nativeimage.cpp

```diff
@@ -285,7 +285,7 @@ void NativeImage::CheckAssemblyMvid(Assembly *assembly) const
     StringFromGUID2(*componentMvid, componentMvidText, MVID_TEXT_LENGTH);
 
     SString message;
-    message.Printf(W("MVID mismatch between loaded assembly '%s' (MVID = %s) and an assembly with the same simple name embedded in the native image '%s' (MVID = %s)"),
+    message.Printf(W("MVID mismatch between loaded assembly '%S' (MVID = %s) and an assembly with the same simple name embedded in the native image '%s' (MVID = %s)"),
```

Comment:
It seems we could also change the native image formatting to `%S` and replace the `SString(SString::Utf8, GetFileName()).GetUnicode()` by just `GetFileName()`. Or maybe the other way - keep the `%s` where you've changed it to `%S` and change the related argument to `SString(SString::Utf8, assembly->GetSimpleName()).GetUnicode()`.  
The reason I am suggesting the other way is that I am not sure if message.Printf can handle UTF-8 strings (maybe it would consider the 8 bit strings to be encoded using something like Windows 1250). Can you please test it with some assembly name that would have non-ascii characters in the name to make sure it handles them properly?
The simpler of way without re-encoding to UTF16 would be preferred if it works.

Reply:
Thanks Jan for pointing that out, I have verified locally that Printf indeed prints complete nonsense when presented with a UTF-8 string. I have modified the code to use Unicode per your suggestion. Based on our offline discussion I also had to update the <code>CrstTypes.def</code> file to declare the fact that the <code>AssemblyLoader</code> lock may now also need to take <code>DebuggerMutex</code> in the case when the MVID validation fails and the application fails fast.